examples/tokens/ERC1155.vy

```diff
@@ -0,0 +1,267 @@
+#      The `_id` argument MUST be the token type being transferred.
+#      The `_value` argument MUST be the number of tokens the holder balance is decreased by and match what the recipient balance is increased by.
+#      When minting/creating tokens, the `_from` argument MUST be set to `0x0` (i.e. zero address).
+#      When burning/destroying tokens, the `_to` argument MUST be set to `0x0` (i.e. zero address).
+# event TransferSingle(address indexed _operator, address indexed _from, address indexed _to, uint256 _id, uint256 _value);
+
+event TransferSingle:
+    _operator: indexed(address)
+    _from: indexed(address)
+    _to: indexed(address)
+    _id: uint256
+    _value: uint256
+
+# @dev Either `TransferSingle` or `TransferBatch` MUST emit when tokens are transferred, including zero value transfers as well as minting or burning (see "Safe Transfer Rules" section of the standard).
+#      The `_operator` argument MUST be the address of an account/contract that is approved to make the transfer (SHOULD be msg.sender).
+#      The `_from` argument MUST be the address of the holder whose balance is decreased.
+#      The `_to` argument MUST be the address of the recipient whose balance is increased.
+#      The `_ids` argument MUST be the list of tokens being transferred.
+#      The `_values` argument MUST be the list of number of tokens (matching the list and order of tokens specified in _ids) the holder balance is decreased by and match what the recipient balance is increased by.
+#      When minting/creating tokens, the `_from` argument MUST be set to `0x0` (i.e. zero address).
+#      When burning/destroying tokens, the `_to` argument MUST be set to `0x0` (i.e. zero address).
+# event TransferBatch(address indexed _operator, address indexed _from, address indexed _to, uint256[] _ids, uint256[] _values);
+
+event TransferBatch:
+    _operator: indexed(address)
+    _from: indexed(address)
+    _to: indexed(address)
+    _ids: uint256[BATCH_SIZE]
+    _value: uint256[BATCH_SIZE]
+
+
+# @dev MUST emit when approval for a second party/operator address to manage all tokens for an owner address is enabled or disabled (absence of an event assumes disabled).
+# event ApprovalForAll(address indexed _owner, address indexed _operator, bool _approved);
+event ApprovalForAll:
+    _owner: indexed(address)
+    _operator: indexed(address)
+    _approved: bool
+
+
+
+supportedInterfaces: HashMap[bytes32, bool]
+
+# https://eips.ethereum.org/EIPS/eip-165
+ERC165_INTERFACE_ID: constant(bytes32)  = 0x0000000000000000000000000000000000000000000000000000000001ffc9a7
+ERC1155_INTERFACE_ID: constant(bytes32) = 0x00000000000000000000000000000000000000000000000000000000d9b67a26
+
+tokensIdCount: uint256
+
+_balanceOf: HashMap[address, HashMap[uint256, uint256]]
+operators: HashMap[address, HashMap[address, bool]]
+
+
+# TODO: decide which batch size to use
+BATCH_SIZE: constant(uint256) = 5
```

Comment:
Can make this bigger, probably want to have a relatively large batch size. maybe 128?